src/OpenApi/src/Services/IDocumentProvider.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Extensions.ApiDescriptions;
+
+/// <summary>
+/// Represents a provider for OpenAPI documents to support build-time generation.
+/// </summary>
+/// <remarks>
+/// The Microsoft.Extensions.ApiDescription.Server package and associated configuration
+/// will execute the `dotnet getdocument` command at build-time to support build-time
+/// generation of documents. The getdocument tool will launch the entry point assembly
+/// and query it for service that implements the `IDocumentProvider` interface. For
+/// historical reasons, the `IDocumentProvider` interface is not exposed publicly from
+/// the framework and the `getdocument` tool will instead query for it using type name.
+/// That means we must declare the `IDocumentProvider` interface under the namespace
+/// that it expects. For more information, see https://github.com/dotnet/aspnetcore/blob/82c9b34d7206ba56ea1d641843e1f2fe6d2a0b1c/src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs#L25.
+/// </remarks>
+internal interface IDocumentProvider
+{
+    IEnumerable<string> GetDocumentNames();
+    Task GenerateAsync(string documentName, TextWriter writer);
```

Comment:
In an ideal world, yes. However.....

Although this `IDocumentProvider` interface is internal, it's public in a sense because it's the same build-time generation hook that [NSwag](https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.AspNetCore/IDocumentProvider.cs) and [Swashbuckle](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/master/src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/DocumentProvider.cs) use. I avoided changing the signature here to avoid introducing any breaks in the experience there. If we wanted to fix this, we'd have to coordinate making the change across all three places and updating the lookup logic [here](https://github.com/dotnet/aspnetcore/blob/63c8031b6a6af5009b3c5bb4291fcc4c32b06b10/src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs).

I think it's worth doing but probably not at the moment. 😞